File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-cd-5-4/PhysicalSettlementPeriod.cpp
// PhysicalSettlementPeriod.cpp 
#include "PhysicalSettlementPeriod.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PhysicalSettlementPeriod::PhysicalSettlementPeriod(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //businessDaysNotSpecifiedNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessDaysNotSpecifiedNode = xmlNode->FirstChildElement("businessDaysNotSpecified");

   if(businessDaysNotSpecifiedNode){businessDaysNotSpecifiedIsNull_ = false;}
   else{businessDaysNotSpecifiedIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessDaysNotSpecifiedNode , address : " << businessDaysNotSpecifiedNode << std::endl;
   #endif
   if(businessDaysNotSpecifiedNode)
   {
      if(businessDaysNotSpecifiedNode->Attribute("href") || businessDaysNotSpecifiedNode->Attribute("id"))
      {
          if(businessDaysNotSpecifiedNode->Attribute("id"))
          {
             businessDaysNotSpecifiedIDRef_ = businessDaysNotSpecifiedNode->Attribute("id");
             businessDaysNotSpecified_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(businessDaysNotSpecifiedNode));
             businessDaysNotSpecified_->setID(businessDaysNotSpecifiedIDRef_);
             IDManager::instance().setID(businessDaysNotSpecifiedIDRef_,businessDaysNotSpecified_);
          }
          else if(businessDaysNotSpecifiedNode->Attribute("href")) { businessDaysNotSpecifiedIDRef_ = businessDaysNotSpecifiedNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessDaysNotSpecified_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(businessDaysNotSpecifiedNode));}
   }

   //businessDaysNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessDaysNode = xmlNode->FirstChildElement("businessDays");

   if(businessDaysNode){businessDaysIsNull_ = false;}
   else{businessDaysIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessDaysNode , address : " << businessDaysNode << std::endl;
   #endif
   if(businessDaysNode)
   {
      if(businessDaysNode->Attribute("href") || businessDaysNode->Attribute("id"))
      {
          if(businessDaysNode->Attribute("id"))
          {
             businessDaysIDRef_ = businessDaysNode->Attribute("id");
             businessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(businessDaysNode));
             businessDays_->setID(businessDaysIDRef_);
             IDManager::instance().setID(businessDaysIDRef_,businessDays_);
          }
          else if(businessDaysNode->Attribute("href")) { businessDaysIDRef_ = businessDaysNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(businessDaysNode));}
   }

   //maximumBusinessDaysNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumBusinessDaysNode = xmlNode->FirstChildElement("maximumBusinessDays");

   if(maximumBusinessDaysNode){maximumBusinessDaysIsNull_ = false;}
   else{maximumBusinessDaysIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumBusinessDaysNode , address : " << maximumBusinessDaysNode << std::endl;
   #endif
   if(maximumBusinessDaysNode)
   {
      if(maximumBusinessDaysNode->Attribute("href") || maximumBusinessDaysNode->Attribute("id"))
      {
          if(maximumBusinessDaysNode->Attribute("id"))
          {
             maximumBusinessDaysIDRef_ = maximumBusinessDaysNode->Attribute("id");
             maximumBusinessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(maximumBusinessDaysNode));
             maximumBusinessDays_->setID(maximumBusinessDaysIDRef_);
             IDManager::instance().setID(maximumBusinessDaysIDRef_,maximumBusinessDays_);
          }
          else if(maximumBusinessDaysNode->Attribute("href")) { maximumBusinessDaysIDRef_ = maximumBusinessDaysNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumBusinessDays_ = boost::shared_ptr<XsdTypeNonNegativeInteger>(new XsdTypeNonNegativeInteger(maximumBusinessDaysNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> PhysicalSettlementPeriod::getBusinessDaysNotSpecified()
{
   if(!this->businessDaysNotSpecifiedIsNull_){
        if(businessDaysNotSpecifiedIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(businessDaysNotSpecifiedIDRef_));
        }else{
             return this->businessDaysNotSpecified_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeNonNegativeInteger> PhysicalSettlementPeriod::getBusinessDays()
{
   if(!this->businessDaysIsNull_){
        if(businessDaysIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNonNegativeInteger>(IDManager::instance().getID(businessDaysIDRef_));
        }else{
             return this->businessDays_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNonNegativeInteger>();
   }
}
boost::shared_ptr<XsdTypeNonNegativeInteger> PhysicalSettlementPeriod::getMaximumBusinessDays()
{
   if(!this->maximumBusinessDaysIsNull_){
        if(maximumBusinessDaysIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNonNegativeInteger>(IDManager::instance().getID(maximumBusinessDaysIDRef_));
        }else{
             return this->maximumBusinessDays_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNonNegativeInteger>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-cd-5-4/PhysicalSettlementPeriod.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PhysicalSettlementPeriod : ISerialized
    {
        public PhysicalSettlementPeriod(XmlNode xmlNode)
        {
            XmlNode businessDaysNotSpecifiedNode = xmlNode.SelectSingleNode("businessDaysNotSpecified");
            
            if (businessDaysNotSpecifiedNode != null)
            {
                if (businessDaysNotSpecifiedNode.Attributes["href"] != null || businessDaysNotSpecifiedNode.Attributes["id"] != null) 
                {
                    if (businessDaysNotSpecifiedNode.Attributes["id"] != null) 
                    {
                        businessDaysNotSpecifiedIDRef_ = businessDaysNotSpecifiedNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(businessDaysNotSpecifiedNode);
                        IDManager.SetID(businessDaysNotSpecifiedIDRef_, ob);
                    }
                    else if (businessDaysNotSpecifiedNode.Attributes["href"] != null)
                    {
                        businessDaysNotSpecifiedIDRef_ = businessDaysNotSpecifiedNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessDaysNotSpecified_ = new XsdTypeBoolean(businessDaysNotSpecifiedNode);
                    }
                }
                else
                {
                    businessDaysNotSpecified_ = new XsdTypeBoolean(businessDaysNotSpecifiedNode);
                }
            }
            
        
            XmlNode businessDaysNode = xmlNode.SelectSingleNode("businessDays");
            
            if (businessDaysNode != null)
            {
                if (businessDaysNode.Attributes["href"] != null || businessDaysNode.Attributes["id"] != null) 
                {
                    if (businessDaysNode.Attributes["id"] != null) 
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["id"].Value;
                        XsdTypeNonNegativeInteger ob = new XsdTypeNonNegativeInteger(businessDaysNode);
                        IDManager.SetID(businessDaysIDRef_, ob);
                    }
                    else if (businessDaysNode.Attributes["href"] != null)
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessDays_ = new XsdTypeNonNegativeInteger(businessDaysNode);
                    }
                }
                else
                {
                    businessDays_ = new XsdTypeNonNegativeInteger(businessDaysNode);
                }
            }
            
        
            XmlNode maximumBusinessDaysNode = xmlNode.SelectSingleNode("maximumBusinessDays");
            
            if (maximumBusinessDaysNode != null)
            {
                if (maximumBusinessDaysNode.Attributes["href"] != null || maximumBusinessDaysNode.Attributes["id"] != null) 
                {
                    if (maximumBusinessDaysNode.Attributes["id"] != null) 
                    {
                        maximumBusinessDaysIDRef_ = maximumBusinessDaysNode.Attributes["id"].Value;
                        XsdTypeNonNegativeInteger ob = new XsdTypeNonNegativeInteger(maximumBusinessDaysNode);
                        IDManager.SetID(maximumBusinessDaysIDRef_, ob);
                    }
                    else if (maximumBusinessDaysNode.Attributes["href"] != null)
                    {
                        maximumBusinessDaysIDRef_ = maximumBusinessDaysNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumBusinessDays_ = new XsdTypeNonNegativeInteger(maximumBusinessDaysNode);
                    }
                }
                else
                {
                    maximumBusinessDays_ = new XsdTypeNonNegativeInteger(maximumBusinessDaysNode);
                }
            }
            
        
        }
        
    
        #region BusinessDaysNotSpecified_
        private XsdTypeBoolean businessDaysNotSpecified_;
        public XsdTypeBoolean BusinessDaysNotSpecified_
        {
            get
            {
                if (this.businessDaysNotSpecified_ != null)
                {
                    return this.businessDaysNotSpecified_; 
                }
                else if (this.businessDaysNotSpecifiedIDRef_ != null)
                {
                    businessDaysNotSpecified_ = IDManager.getID(businessDaysNotSpecifiedIDRef_) as XsdTypeBoolean;
                    return this.businessDaysNotSpecified_; 
                }
                else
                {
                      return this.businessDaysNotSpecified_; 
                }
            }
            set
            {
                if (this.businessDaysNotSpecified_ != value)
                {
                    this.businessDaysNotSpecified_ = value;
                }
            }
        }
        #endregion
        
        public string businessDaysNotSpecifiedIDRef_ { get; set; }
        #region BusinessDays_
        private XsdTypeNonNegativeInteger businessDays_;
        public XsdTypeNonNegativeInteger BusinessDays_
        {
            get
            {
                if (this.businessDays_ != null)
                {
                    return this.businessDays_; 
                }
                else if (this.businessDaysIDRef_ != null)
                {
                    businessDays_ = IDManager.getID(businessDaysIDRef_) as XsdTypeNonNegativeInteger;
                    return this.businessDays_; 
                }
                else
                {
                      return this.businessDays_; 
                }
            }
            set
            {
                if (this.businessDays_ != value)
                {
                    this.businessDays_ = value;
                }
            }
        }
        #endregion
        
        public string businessDaysIDRef_ { get; set; }
        #region MaximumBusinessDays_
        private XsdTypeNonNegativeInteger maximumBusinessDays_;
        public XsdTypeNonNegativeInteger MaximumBusinessDays_
        {
            get
            {
                if (this.maximumBusinessDays_ != null)
                {
                    return this.maximumBusinessDays_; 
                }
                else if (this.maximumBusinessDaysIDRef_ != null)
                {
                    maximumBusinessDays_ = IDManager.getID(maximumBusinessDaysIDRef_) as XsdTypeNonNegativeInteger;
                    return this.maximumBusinessDays_; 
                }
                else
                {
                      return this.maximumBusinessDays_; 
                }
            }
            set
            {
                if (this.maximumBusinessDays_ != value)
                {
                    this.maximumBusinessDays_ = value;
                }
            }
        }
        #endregion
        
        public string maximumBusinessDaysIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

